cpp/ql/src/semmle/code/cpp/controlflow/internal/ConstantExprs.qll

```diff
@@ -104,9 +104,42 @@ private predicate loopConditionAlwaysUponEntry(ControlFlowNode loop, Expr condit
   )
 }
 
+/**
+ * This relation is the same as the `el instanceof Function`, only obfuscated
+ * so the optimizer will not understand that any `FunctionCall.getTarget()`
+ * should be in this relation.
+ */
+pragma[noinline]
+private predicate isFunction(Element el) {
+  el instanceof Function
+  or
+  el.(Expr).getParent() = el
+}
+
+/**
+ * Holds if `fc` is a `FunctionCall` with no return value for `getTarget`. This
+ * can happen due to extractor issue CPP-383.
+ */
+pragma[noopt]
+private predicate callHasNoTarget(@funbindexpr fc) {
+  exists(Function f |
+    funbind(fc, f) and
+    not isFunction(f)
+  )
+}
+
+// This base case is pulled out to work around QL-796
```

Comment:
Do we have a GH issue for this instead?

Reply:
We do now. I've updated all references to QL-796. We [decided on Slack](https://github.slack.com/archives/CPNQ6CE1L/p1618928140255400?thread_ts=1618927609.252500&cid=CPNQ6CE1L) that it's okay to link to this internal issue from code.